sources/packages/api/src/route-controllers/report/report.aest.controller.ts

```diff
@@ -0,0 +1,144 @@
+import { ClientTypeBaseRoute } from "../../types";
+import { StringBuilder } from "../../utilities/string-builder";
+import {
+  getDateOnlyFormatAsString,
+  getFileNameTimestamp,
+  CustomNamedError,
+} from "../../utilities";
+import BaseController from "../BaseController";
+import { ReportsFilterAPIInDTO } from "./models/report.dto";
+import { FormNames } from "../../services/form/constants";
+import {
+  REPORT_CONFIG_NOT_FOUND,
+  FILTER_PARAMS_MISMATCH,
+} from "../../services/report/constants";
+
+/**
+ * Controller for Reports for AEST Client.
+ * This consists of all Rest APIs for notes.
+ */
+@AllowAuthorizedParty(AuthorizedParties.aest)
+@Groups(UserGroups.AESTUser)
+@Controller("reports")
+@ApiTags(`${ClientTypeBaseRoute.AEST}-reports`)
+export class ReportAESTController extends BaseController {
+  constructor(
+    private readonly reportService: ReportService,
+    private readonly formService: FormService,
+  ) {
+    super();
+  }
+
+  /**
+   * Rest API to financial reports in csv format.
+   * @param payload report filter payload.
+   * @param response
+   */
+  @ApiBadRequestResponse({
+    description: "Not able to export report due to an invalid request.",
+  })
+  @ApiUnprocessableEntityResponse({
+    description:
+      "Either report config missing or filter parameters not matching the report config.",
+  })
+  @Post()
+  async exportReport(
+    @Body() payload: ReportsFilterAPIInDTO,
+    @Res() response: Response,
+  ): Promise<void> {
+    const submissionResult = await this.formService.dryRunSubmission(
+      FormNames.ExportFinancialReports,
+      payload,
+    );
+
+    if (!submissionResult.valid) {
+      throw new BadRequestException(
+        "Not able to export report due to an invalid request.",
+      );
+    }
+
+    let reportData = [];
+
+    try {
+      reportData = await this.reportService.exportReport(payload);
+    } catch (error: unknown) {
+      if (error instanceof CustomNamedError) {
+        switch (error.name) {
+          case REPORT_CONFIG_NOT_FOUND:
+          case FILTER_PARAMS_MISMATCH:
+            throw new UnprocessableEntityException(error.message);
+          default:
+            throw error;
+        }
+      }
+      throw error;
+    }
+
+    if (!reportData || reportData.length === 0) {
+      this.streamFile(response, payload.reportName, "No data found.");
+      return;
+    }
+
+    const reportCSVContent = new StringBuilder();
+    const reportHeaders = Object.keys(reportData[0]);
+    reportCSVContent.appendLine(reportHeaders.join(","));
+
+    reportData.forEach((reportDataItem) => {
+      let dataItem = "";
+      reportHeaders.forEach((header, index) => {
+        const data =
+          reportDataItem[header] instanceof Date
+            ? getDateOnlyFormatAsString(reportDataItem[header])
+            : reportDataItem[header];
+        dataItem += index ? `,${data}` : data;
+      });
+      reportCSVContent.appendLine(dataItem);
```

Comment:
I believe that the logic to have a data set converted to a CSV belongs more to the service than the controller.

Reply:
I quite disagree with this statement: "Data presented as CSV is a presentation-layer kinda responsibility".
Getting the CSV and writing it to the HTTP response is an API controller-specific task, the CSV generation is an IU-related task as much as an e-Cert generation, IMO.

Reply:
Let me describe that in a better way to be clear. 
My point is -> in API layer, transforming the data in a format that is required for any client is controller's responsibility than than service's.

Reply:
Still not on the same page here.
@ann-aot @guru-aot @henryoforeh-aot any thoughts?

Reply:
I also commented something similar, IMO, file logic should not be inside the controller.

Reply:
I agree as well, I think the controller should not have logic.

Reply:
IMO and understanding, The service layer fetches and transforms the data from the db, like all the heavy loading, the controller connects the service layer and exports/imports the data from/to the end point. I believe we try to always keep the controller simple on the data perspective.